packages/plugins/src/plugins/glint-fix.plugin.ts

```diff
@@ -0,0 +1,93 @@
+import { applyCodeFix, makeCodeFixStrict } from '@rehearsal/codefixes';
+import {
+  GlintService,
+  Plugin,
+  PluginOptions,
+  PluginsRunnerContext,
+  type PluginResult,
+} from '@rehearsal/service';
+import debug from 'debug';
+
+import { pathUtils } from '@glint/core';
+import { CodeFixAction } from 'typescript';
+import { CodeActionKind, Diagnostic } from 'vscode-languageserver';
+
+const DEBUG_CALLBACK = debug('rehearsal:plugins:glint-fix');
+
+export class GlintFixPlugin implements Plugin<PluginOptions> {
+  async run(fileName: string, context: PluginsRunnerContext): PluginResult {
+    const allFixedFiles: Set<string> = new Set();
+    const service = context.service as GlintService;
+
+    let diagnostics = this.getDiagnostics(service, fileName);
+
+    while (diagnostics.length > 0) {
+      const diagnostic = diagnostics.shift();
+
+      if (diagnostic === undefined) {
+        break;
+      }
+      const fileText = service.getFileText(fileName);
+      const start = pathUtils.positionToOffset(fileText, diagnostic.range.start);
+
+      const fix = this.getCodeFix(fileName, diagnostic, service);
+
+      if (fix === undefined) {
+        continue;
+      }
+
+      applyCodeFix(fix, {
+        getText(filename: string) {
+          return context.service.getFileText(filename);
+        },
+
+        applyText(newText: string) {
+          DEBUG_CALLBACK(`- TS${diagnostic.code} at ${start}:\t ${newText}`);
+        },
+
+        setText(filename: string, text: string) {
+          context.service.setFileText(filename, text);
+          allFixedFiles.add(filename);
+          DEBUG_CALLBACK(`- TS${diagnostic.code} at ${start}:\t codefix applied`);
+        },
+      });
+
+      // Get updated list of diagnostics
+      diagnostics = this.getDiagnostics(service, fileName);
+    }
+
+    return Promise.resolve(Array.from(allFixedFiles));
+  }
+
+  getDiagnostics(service: GlintService, fileName: string): Diagnostic[] {
+    return service.getDiagnostics(fileName).map((d) => service.convertTsDiagnosticToLSP(d));
+  }
+
+  getCodeFix(
+    fileName: string,
+    diagnostic: Diagnostic,
+    service: GlintService
+  ): CodeFixAction | undefined {
+    const glintService = service.getGlintService();
```

Comment:
Your work looks awesome! 

But I'm curious why does diagnostic plugin was not used directly?
The logic of converting Glint codefixes to TS codefixes cab be hidden inside service.getDiagnostics + logic to applying them can be moved to both services as well, so:

- rehearsal.getCodeFix and gling.getCodeFixes (includes positionToOffset) return the same format
- diagnostic-fix just called geDiagnostic -> getCodeFixes -> applies the same way for both services

Reply:
Good question. Given the subtle differences between applying codefixes from Glint vs from Rehearsal it felt more straightforward to just separate them and have a separate plugin for gjs/hbs files rather than trying to conditionally handle everything in one plugin.